ONSdigital/rm-sdx-gateway
# Update to Java 11

# Motivation and Context
Updating JRE from 8 to 11. The resulting image size has grown only 12 MB for this. J9+ modularisation will allow us to customise the image if required, using either our own base image or a multistage build.


# What has changed
Updated version of parent image, updated version in GH Actions, added jaxb and javax.annotations as dependencies to POM, updated a few lib dependencies, and replaced abandonware Cobertura with JaCoCo 





# How to test?
- `mvn clean install`
- `mvn spring-boot:run`
- deploy gateway with new image to a sandbox cluster
- run acceptance tests on sandbox cluster 
- set port forwarding from localhost with ras-rm-cli
- `curl -u $USER:$PASSWD -H "Accept: application/json" -H "Content-Type: application/json" http://localhost:8191/receipts -v -d "{\"userId\":\"ded66326-6365-4eca-afca-12bd2aca7b9b\",\"caseId\": \"069e61fe-8e87-4b50-8af9-3c4a34a643bf\"}"`
- check gateway pod logs for errors, and check Case.Responses queue in RabbitMQ (localhost:15672)
- check that receipting works in frontstage (localhost:8082) and response-ops-ui (localhost:8085)




# Links
- https://trello.com/c/H1bpRy5O
